### ISSUE TITLE
Confused About Code Calc AvgLoss

### ISSUE THREAD

**Issue by Vela-zz** [awsm-research/LineVul#10]
I am just confused about this line,  why use np.exp to calc a avg_loss ?  and logging_loss and tr_nb seems to always 0?
` avg_loss=round(np.exp((tr_loss - logging_loss) /(global_step- tr_nb)),4)
`
and why this section train_loss do not multiply args.gradient_accumulation_steps?

` avg_loss = round(train_loss/tr_num,5)`
most of code about train/test is similiar with microsoft/codebert in this repo, but codeBERT just calc their loss below,

```
        nb_tr_examples, nb_tr_steps,tr_loss,global_step,best_bleu,best_loss = 0, 0,0,0,0,1e6 
        bar = tqdm(range(num_train_optimization_steps),total=num_train_optimization_steps)
        train_dataloader=cycle(train_dataloader)
        eval_flag = True
        for step in bar:
            batch = next(train_dataloader)
            batch = tuple(t.to(device) for t in batch)
            source_ids,source_mask,target_ids,target_mask = batch
            loss,_,_ = model(source_ids=source_ids,source_mask=source_mask,target_ids=target_ids,target_mask=target_mask)
            
            if args.n_gpu > 1:
                loss = loss.mean() # mean() to average on multi-gpu.
            if args.gradient_accumulation_steps > 1:
                loss = loss / args.gradient_accumulation_steps
            tr_loss += loss.item()
            train_loss=round(tr_loss*args.gradient_accumulation_steps/(nb_tr_steps+1),4)
```

**Comment by MichaelFu1998-create** [comment]
Hi @Vela-zz , please ignore `avg_loss=round(np.exp((tr_loss - logging_loss) /(global_step- tr_nb)),4)` as the correct `avg_loss` is calculated as follows in each training step:
```
avg_loss = round(train_loss/tr_num,5)
bar.set_description("epoch {} loss {}".format(idx,avg_loss))
```

`train_loss` does not multiply `args.gradient_accumulation_steps` as we accumulate the loss value in each step via
`train_loss += loss.item()`

Hope this helps.

**Comment by Vela-zz** [comment]
Thanks for your help
